### PR TITLE
Upgrade to CircleCI 2.0

### PR DESCRIPTION
**Pull request by autero1**
This PR upgrades CircleCI configuration to 2.0 to re-enable tests. As the new configuration runs `terraform fmt` and fails the build if `fmt` is required, some of the changed files are a result of formatting the modules.

### CODE REVIEW

**Comment by autero1** [comment]
For some reason @Etiene wasn't added as reviewer, even though she's in the `CODEOWNERS` file. Cannot add her manually, either.

**Comment by brikis98** [comment]
> For some reason @Etiene wasn't added as reviewer, even though she's in the CODEOWNERS file. Cannot add her manually, either.

She hasn't accepted the invite to the repo that gives her permissions

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+workspace_root: &workspace_root
+                  /go/src/github.com/hashicorp/terraform-aws-nomad
```
Comment (brikis98) [inline .circleci/config.yml]: Indentation seems weird?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+version: 2
+jobs:
+  validate_terraform:
+    docker:
+      - image: hashicorp/terraform
+    steps:
+      - checkout
+      - run:
+          name: Validate Terraform Formatting
+          command: "[ -z \"$(terraform fmt -write=false)\" ] || { terraform fmt -write=false -diff; exit 1; }"
```
Comment (brikis98) [inline .circleci/config.yml]: Interesting approach! We also have [pre-commit hooks](https://github.com/gruntwork-io/pre-commit) for some of this stuff.

Comment (autero1) [reply .circleci/config.yml]: Ah ok, I sought inspiration from the `terraform-aws-consul` -repo and though this was the way to go with the Hashicorp -modules.

**Comment by autero1** [comment]
One additional thing... Test `TestNomadConsulClusterColocatedWithUbuntuAmi` failed once during `packer` build with this:
```
Error launching source instance: Unsupported: The requested configuration is currently not supported. Please check the documentation for supported configurations.
``` 
I've only seen this once, when the region was `eu-north-1`. 

Is this something we should look into?

**Comment by brikis98** [comment]
IIRC, `eu-north-1` has no `t2` instances. I've mostly just been excluding it from the regions we use for tests as a workaround. I suppose an alternative is to switch to `t3` instances, but it's likely some regions don't have those either, so excluding seems like a simpler solution for now.

**Pull request merged**